### ISSUE TITLE
conver python dict() to RedisDict

### ISSUE THREAD

**Issue by Sorrow2** [brainix/pottery#5]
is there any way not to add key, value in loop and do it like this :
`ned = RedisDict(redis=redis, key={'key': 'val'})`
Following code throws:
`
Exception ignored in: <bound method RedisDict.__del__ of RedisDict{}>
Traceback (most recent call last):
  File "/usr/local/lib/python3.4/dist-packages/pottery/base.py", line 48, in __del__
    if self.key.startswith(self._RANDOM_KEY_PREFIX):
AttributeError: 'dict' object has no attribute 'startswith'
`

**Comment by brainix** [comment]
Hello!

If I understand your question, you're trying to populate your `RedisDict` at instantiation?  The `key` argument represents the Redis key itself, not the contents of the `RedisDict`.  I believe that this is what you're trying to do:

`>>> ned = RedisDict({'key': 'val'}, redis=redis, key='ned')`

This way, you instantiate a `RedisDict` with the Redis key `ned`, and the contents `{'key': 'val'}`.  Does that answer your question?  Thanks for the feedback!

**Comment by Sorrow2** [comment]
Thanks.
What i want to do is to overwrite one dict with other, like this:
```
>>> x_old = {1: 'olddata'}
>>> x_new = {1: 'newdata'}
>>> x_old = x_new
>>> print(x_old)
{1: 'newdata'}
```
So in pottery i want all data in RedisDict be consistently (every x seconds) replaced with data in usual python dict.
Your example:
```
tmpdict = {'key': 'val', 'key2': 'val2'}
ned = RedisDict(tmpdict, redis=redis, key='neds')
```
works, but only once. On second iteration i get:
```
  File "/usr/local/lib/python3.4/dist-packages/pottery/dict.py", line 29, in __init__
    raise KeyExistsError(self.redis, self.key)
pottery.exceptions.KeyExistsError: neds
```
I can use 
```
 x_old = RedisDict(redis=redis, key='neds')

for k, v in x_new.items():
    x_old[k] = v
```
But i am wondering if there is a better way?

**Comment by brainix** [comment]
Ah, interesting use case.  Let me think about the best way to handle this, and I'll get back to you tonight.

**Comment by brainix** [comment]
A thought...  You could do this:

```
>>> tmpdict = {'key': 'val', 'key2': 'val2'}
>>> redis.delete('neds')
>>> ned = RedisDict(tmpdict, redis=redis, key='neds')
```

That would work for multiple iterations.  The reason that you get the `KeyExistsError` is to prevent you from accidentally overwriting an existing Redis key.  Does that work for you?

**Comment by Sorrow2** [comment]
Yeah, thats works.
Couple more questions. I am new to redis.
If i just add key to redis, it will be never deleted until i delete it manually? How can i delete database from pottery (like FLUSHDB)? 
Is there way to make temporary keys with pottery?

**Comment by brainix** [comment]
Great!  Ok, let me try to answer your questions:

1.  If you add a key to Redis, by default, it will never be deleted until you delete it manually.  That is, unless you specify an expiration time.  You can set an expiration time like this:

    `>>> redis.expire('key', 10)`

    This command means that Redis will expire your key in 10 seconds (as the expiration time is specified in seconds).

2.  The easiest way to delete your entire Redis database is:

    `>>> redis.flushall()`

3.  You can make temporary keys with Pottery.  Just instantiate your `RedisSet`, `RedisDict`, or `RedisList` without specifying a key like so:

    `>>> tmp = RedisDict(redis=redis)`

    Pottery will assign its own temporary key, and Pottery will also delete your temporary `RedisDict` as soon as it goes out of scope.

Please let me know if that helps!  Good luck!

**Comment by Sorrow2** [comment]
That helps. Thank you

**Comment by brainix** [comment]
No problem.  Closing out this issue now, but please feel free to create another issue if you run into another problem.
